packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -310,9 +321,25 @@ abstract class ScrollView extends StatelessWidget {
         return buildViewport(context, offset, axisDirection, slivers);
       },
     );
-    return primary && scrollController != null
-      ? PrimaryScrollController.none(child: scrollable)
-      : scrollable;
+    final Widget scrollableResult = primary && scrollController != null
+        ? PrimaryScrollController.none(child: scrollable)
+        : scrollable;
+
+    if (keyboardDismissBehavior == ScrollViewKeyboardDismissBehavior.onDrag) {
+      final FocusNode dismissNode = FocusNode();
```

Comment:
You don't want to recreate this `FocusNode` every build, because you need to dispose of it, which (sadly) means you'd need to make this a stateful widget.

But... What's your goal here? Do you just want to know if a context doesn't have focus?  If so, you could use something like this, and keep it stateless.

```dart
  return Focus(
    child: Builder(builder: (BuildContext context) {
      return NotificationListener<ScrollUpdateNotification>(
        child: scrollableResult,
        onNotification: (ScrollUpdateNotification notification) {
          final FocusNode node = Focus.of(context);
          if (notification.dragDetails != null && !node.hasFocus) {
            node.requestFocus();
          }
          return true;
        },
      );
    }),
  );
```

Reply:
For the focus node, I'm using that to dismiss the keyboard when scrolling happens.  I don't want to spam the Focus every time a scroll event happens so I keep a reference to the dismissNode to make sure that I only requestFocus once.  It's fine if the FocusNode gets disposed, it's just a way to filter out duplicate events, if it never worked everything would still look the same.

Reply:
Where are you disposing it?

Reply:
I didn't realize we were talking about literally calling `.dispose()`.  I'll rework it.

Reply:
Started using unfocus instead so no FocusNode instance is needed.